app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -716,7 +728,7 @@ private void returnWithResult(Intent intent) {
     public void afterTextChanged(Editable incomingEditable) {
         final String incomingString = incomingEditable.toString();
         final String currentSearchText = getSearchText().toString();
-        if (!"".equals(currentSearchText) && incomingString.equals(currentSearchText)) {

```

Comment:
this made it so that when you changed the input from 1 character to none, that change would not register in any way